scipy/stats/_distribution_infrastructure.py

```diff
@@ -1150,44 +1184,32 @@ def wrapped(self, x, y, *args, **kwargs):
         dtype = np.result_type(x.dtype, y.dtype, self._dtype)
         # yes, copy to avoid modifying input arrays
         x, y = x.astype(dtype, copy=True), y.astype(dtype, copy=True)
+        all_ood = (x < low) & (y < low) | (x > high) & (y > high)
+        swap = x > y
 
-        # Swap arguments to ensure that x < y, and replace
-        # out-of domain arguments with domain endpoints. We'll
-        # transform the result later.
-        i_swap = y < x
-        x[i_swap], y[i_swap] = y[i_swap], x[i_swap]
-        i = x < low
-        x[i] = low[i]
-        i = y < low
-        y[i] = low[i]
-        i = x > high
-        x[i] = high[i]
-        i = y > high
-        y[i] = high[i]
+        x[x < low] = low[x < low]
+        x[x > high] = high[x > high]
+        y[y > high] = high[y > high]
+        y[y < low] = low[y < low]
```

Comment:
I imagine this grew out of another diff to end back at the same logic, but it is just slower than what was here. I'll come back to see if it can be improved, though, so for now let's leave it be.